File: packages/core/android/src/main/java/com/reactnativedrivekitcore/UserInfoMappers.kt
package com.reactnativedrivekitcore

import com.drivequant.drivekit.core.driver.UserInfo
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReadableMap

class UserInfoMappers {
  companion object {
    fun mapUserInfoToReadableMap(userInfo: UserInfo?): ReadableMap? {
      if(userInfo == null) {
        return null;
      }
      var userInfoMap = Arguments.createMap();
      userInfoMap.putString("pseudo", userInfo.pseudo);
      userInfoMap.putString("firstname", userInfo.firstname);
      userInfoMap.putString("lastname", userInfo.lastname);
      return userInfoMap;
    }
  }
}

File: packages/core/ios/UserInfoMappers.swift
//
//  UserInfoMappers.swift
//  react-native-drivekit-core
//
//  Created by Cyril Bonaccini on 10/11/2022.
//

import Foundation
import DriveKitCoreModule


func mapUserInfoToNSDictionary (userInfo: UserInfo?) -> NSDictionary? {
    if let info = userInfo {
        return [
            "firstname" : info.firstname as Any,
            "lastname" : info.lastname as Any,
            "pseudo" : info.pseudo as Any,
        ]
    } else {
        return nil;
    }
}
